### PR TITLE
refactor(policy-pack): split mdc_to_pack_mapping_test.clj — extract fields (2/7)

### PR DESCRIPTION
**Pull request by H31MDALLR**
# refactor(policy-pack): split mdc_to_pack_mapping_test.clj — extract fields (2/7)

## Overview

Slice 2 of the 7-PR split of
`components/policy-pack/test/ai/miniforge/policy_pack/mdc_to_pack_mapping_test.clj`
(stratum-lint SL003, rule 210, `standards/miniforge`). See
[slice 1, miniforge#1758](https://github.com/miniforge-ai/miniforge/pull/1758)
for the full train rationale and precedent links.

This slice extracts `ai.miniforge.policy-pack.mdc-to-pack-mapping-test.fields`:
six independent, dependency-free field-derivation helpers —
`derive-description`, `normalize-globs`, `build-always-inject`,
`build-enforcement`, `extract-agent-behavior-section`,
`extract-first-paragraph`. None of these call each other or anything
else in the parent namespace, so unlike slice 1's dewey chain this was
always a layer-0 island — moving it out shrinks the parent file's bulk
without shortening its deepest same-file call chain.

## Motivation

Zero fan-in confirmed for the parent test namespace before slice 1
(see miniforge#1758); this slice doesn't change that — no other
namespace requires `mdc-to-pack-mapping-test` or its `fields` sibling.

## Changes in Detail

- New file `mdc_to_pack_mapping_test/fields.clj`: the six helpers
  above, all layer 0, moved verbatim (only their `:stratum` tags and
  the file's own header/docstring are new).
- New file `mdc_to_pack_mapping_test/fields_test.clj`: the six
  `deftest` forms that exercise them (`description-generation-test`,
  `always-inject-mapping-test`, `globs-normalization-test`,
  `agent-behavior-section-extraction-test`,
  `first-paragraph-extraction-test`,
  `edge-case-body-only-headings-test`) — assertions unchanged, call
  sites now go through the `fields` alias.
- `mdc_to_pack_mapping_test.clj`: the six functions and six tests
  removed; `build-applies-to` and `compile-rule` now call
  `fields/normalize-globs`, `fields/derive-description`,
  `fields/build-enforcement`, and `fields/build-always-inject` across
  the namespace boundary. Ran `stratum-lint --fix` to renumber the
  remaining defs' `;--- Layer N` headings and `^{:stratum n}`
  metadata — still 4 real layers (unchanged by this slice, as noted
  above; the deepest chain is now `dewey`/`build-applies-to` →
  `derive-title`/`compile-rule` → the `compile-rule`-dependent tests,
  which slices 4 and 6 resolve).

The parent namespace stays over budget (4 real layers) until the
remaining slices land; the commit removing the `fields` helpers used
`MINIFORGE_STRATUM_BUDGET_MODE=warn`, same convention as slice 1.

## Testing Plan

1. `clj-kondo` clean on all three touched/new files.
2. stratum-lint: `fields.clj` and `fields_test.clj` pass SL003
   outright; `mdc_to_pack_mapping_test.clj` intentionally still over
   budget (4 layers) — expected and tracked, not a defect.
3. Test/assertion count verified unchanged before and after, across
   all three namespaces combined: 39 tests / 1213 assertions, 0
   failures — identical to the slice-1 baseline.
4. Full pre-commit suite (`poly:check`, lint, stratum-lint, smoke
   tests, GraalVM) passed on both commits.
5. Adversarial self-review: diffed the full top-level `defn`/`deftest`
   set before and after — exactly 6 functions and 6 deftest forms
   relocated, 0 added/removed/altered in behavior; the four remaining
   call sites (`build-applies-to`, `compile-rule`) changed only their
   qualification (`fields/...`), not their arguments or logic.

PR size: 279 reportable lines (259 raw insertions / 191 raw deletions
across 3 files, two commits of 149 and 171 reportable lines each),
under the 600-line budget — no override needed.

## Deployment Plan

Merges to `main` as part of the ongoing 7-PR train. The next slice
rebases onto the updated `main` after this one merges.

## Related Issues/PRs

- Slice 1: [miniforge#1758](https://github.com/miniforge-ai/miniforge/pull/1758)
- Precedent: [mdc_compiler.clj split train, miniforge#1729-#1743](https://github.com/miniforge-ai/miniforge/pull/1743)
- Part of the stratum-lint rule-210 remediation program (Wave 2, policy-pack batch 2)

## Checklist

- [x] Zero fan-in confirmed (unchanged since slice 1)
- [x] Pure code motion — no behavior change, no assertions altered
- [x] `clj-kondo` clean
- [x] Tests green (39/39 tests, 1213/1213 assertions, matches baseline)
- [x] PR-diff and commit-diff budgets checked (279/600 PR, both commits ≤200)
- [x] `MINIFORGE_STRATUM_BUDGET_MODE=warn` used + documented for the
      expected intermediate over-budget state

### CODE REVIEW

File: components/policy-pack/test/ai/miniforge/policy_pack/mdc_to_pack_mapping_test.clj [hunk 1]
```diff
@@ -451,7 +314,7 @@
   [{:keys [filepath frontmatter body]}]
   (let [dewey       (get frontmatter "dewey" "000")
         title       (derive-title frontmatter filepath)
-        description (derive-description dewey title)
+        description (fields/derive-description dewey title)
         always-apply (get frontmatter "alwaysApply")
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/policy-pack/test/ai/miniforge/policy_pack/mdc_to_pack_mapping_test.clj:318]: Pre-existing behavior, unchanged by this slice — this is a verbatim move of `compile-rule`'s `always-apply` handling from the original single-file namespace (only the `fields/build-enforcement` / `fields/build-always-inject` call qualification changed, not the `(get frontmatter "alwaysApply")` line itself or its consumers). This split train is scoped to pure code motion — no logic changes — per the repo's rule-210 split convention (see miniforge#1729-#1743). Agreed this looks like a real acceptance-suite/production divergence worth fixing; flagging it as a follow-up rather than folding it into this mechanical PR.

File: components/policy-pack/test/ai/miniforge/policy_pack/mdc_to_pack_mapping_test.clj [hunk 2]
```diff
@@ -191,6 +122,15 @@
...
           "All frontmatter fields must have mappings"))))
 
+(defn ^{:stratum 0} build-applies-to
+  "Build :rule/applies-to from dewey + optional globs."
+  [dewey-str globs]
+  (let [phases (dewey/dewey-to-phases dewey-str)
+        base   {:phases phases}]
+    (if-let [g (fields/normalize-globs globs)]
+      (assoc base :file-globs g)
+      base)))
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/policy-pack/test/ai/miniforge/policy_pack/mdc_to_pack_mapping_test.clj:132]: Pre-existing behavior, unchanged by this slice — this is a verbatim move of `build-applies-to`'s `if-let` logic from the original single-file namespace (only the `fields/normalize-globs` call qualification changed). Same as the sibling comment on `always-apply`: this split train is scoped to pure code motion, so I'm not altering reference-implementation logic here. Agreed it's a real divergence from `mdc_compiler.clj`'s `(seq globs)` gating and worth a follow-up.
